packages/errors/package.json

```diff
@@ -64,6 +64,9 @@
     "engine": {
         "node": ">=17.4"
     },
+    "dependencies": {
+        "@solana/addresses": "workspace:*"
+    },
```

Comment:
Wait what? Is this allowed?

I understand this is needed to define the context of some errors but IMO, **the `@solana/errors` package should be a leaf dependency** to avoid any circular dependencies. If we start adding other packages from the monorepo as dependencies here we're shooting ourselves in the foot because just about every single package of the monorepo will end-up depending on `@solana/errors`.

I wonder if maybe there's a case for each packages to define their own context then. 🤔

Reply:
I don't know. Does it still count as a circular dependency (in the sense that it's consequential) if the error module is only pulling in _types_ from all the other packages, and not implementations? I didn't actually try to add an error to `@solana/addresses` yet, which would answer my question.

Two options here: 

1. Make the context structurally similar (ie. `string[]` instead of `Address[]`). This would make the context less rich and less useful in a `catch` block (ie. you'd get an array of `string` for a `TRANSACTION_SIGNATURE_MISSING` instead of an array of `Address`
2. Evict the `Address` type to some other package.

Thoughts?

Reply:
Yeah, you're 100% correct this is a problem.

```shell
~/src/solana-web3.js-git$ pnpm turbo compile:js compile:typedefs test:typecheck test:unit:node
 ERROR  run failed: Invalid package dependency graph: cyclic dependency detected:
        @solana/errors, @solana/addresses
```

Reply:
OK. I tried some more stuff but there's no way out. Should we throw in the towel and make this `string[]`, move `Address` somewhere else, or a third thing that I haven't thought of yet?

Reply:
I'm not totally disgusted by `string[]` instead of `Address[]`. I also think that by ~~forcing~~ encouraging JS primitives to be used in the context, it makes its serialization more robust and it makes it less likely that templated strings for error message won't be sufficient.

Reply:
Yeah, at one point I thought about somehow constraining the type of `SolanaErrorContext` such that it has to be made up of primitives or objects/arrays of primitives. This makes context _so_ much less useful though, because now you can't do this sort of thing:

```ts
try {
  /* ... */
} catch (e) {
    if (isSolanaError(e, SOLANA_ERROR__TRANSACTION_SIGNATURE_MISSING) {
        const noopSigners = e.context.addresess.map(createNoopSigner);
        // Retry…
    }
}
```

Because `e.context.addresess` is `string[]` instead of `Address[]`. You'd have to do extra work like…

```ts
if (isSolanaError(e, SOLANA_ERROR__TRANSACTION_SIGNATURE_MISSING) {
    const noopSigners = e.context.addresess.map(putativeAddress => {
        assertIsAddress(putativeAddress);
        return createNoopSigner(address);
    });
    // Retry…
}
```

Maybe that's a good thing.